### PR TITLE
Story/vspc 180

### PR DESCRIPTION
**Pull request by glend17**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 

https://diging.atlassian.net/browse/VSPC-180
  
## Anything else the reviewer needs to know?

No

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java [hunk 1]
```diff
@@ -59,6 +64,12 @@ public String space(@PathVariable("id") String id, Model model) {
         if (space.getSpaceStatus() == null || space.getSpaceStatus().equals(SpaceStatus.PUBLISHED)
                 || authenticationFacade.getAuthenticatedUser() != null) {
             IExhibition exhibition = exhibitManager.getStartExhibition();
+            List<ExhibitionAboutPage> aboutPageList = aboutPageManager.findAll();
+            ExhibitionAboutPage exhibitionAboutPage = aboutPageList != null && !aboutPageList.isEmpty() ? aboutPageList.get(0):new ExhibitionAboutPage();
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java:68]: Let's add a method to the aboutPageManager that returns the about page; once we add multiple languages support this will likely stop working, so putting that logic into its own manager method will make it easier to adjust.

File: vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java [hunk 2]
```diff
@@ -59,6 +64,12 @@ public String space(@PathVariable("id") String id, Model model) {
...
+            if(exhibitionAboutPage != null) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java:69]: can that ever be the case?

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html [hunk 3]
```diff
@@ -44,6 +70,11 @@
 		        </div>
 		      </div>
 		    </nav>
+		    <div id="aboutContent" th:if="${aboutPageTitle!=null}">
+                        <span class="icon-circle-close closebtn" th:onclick="closeExhibitionDescription()"></span>
+                        <div class="aboutDescription">[[${aboutPageTitle}]]</div>
+                        <div th:if="aboutPageText != null &&!#strings.isEmpty(aboutPageText)} " class="aboutDescription">[[${aboutPageText}]]</div>
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:76]: The page doesn't load on staging. My guess is that there is an opening `{` missing here.

**Comment by jdamerow** [comment]
The link to show the about text should be an info icon (right now it just says "About", which I assume is the title?)

Also for the next PR, please copy the ticket description not just the link to the ticket in the header.

File: vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java (hunk 2 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java:69]: When the about page is not configured exhibitionAboutPage can be null.

File: vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java (hunk 1 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java:68]: Done

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html (hunk 3 above)
Comment (glend17) [reply vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:76]: Yes, have fixed this now

File: vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java (hunk 2 above)
Comment (jdamerow) [reply vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java:68]: but line 68 initializes the variable in that case, doesn't it?

Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java:68]: Sorry I meant exhibitionAboutPage.getTitle() and exhibitionAboutPage.getAboutPageText() will be null if the about page is not configured so I thought there is no need to add that to the model attributes. But I am checking null anyway in the html so I can remove the null check here, I will make the change.

Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSpaceController.java]: Yes my bad. I see what you mean. I have made the change

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
> The link to show the about text should be an info icon

Also, the about page should have its own page, not just a small pop up.

**Comment by glend17** [comment]
> > The link to show the about text should be an info icon
> 
> Also, the about page should have its own page, not just a small pop up.

Have now created a new About Page and also added an info icon

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java [hunk 4]
```diff
@@ -41,4 +41,11 @@ public ExhibitionAboutPage store(ExhibitionAboutPage exhibitionAboutPage) {
         return repo.save(exhibitionAboutPage);
     }
     
+    @Override
+    public ExhibitionAboutPage getExhibitionAboutPage() {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:45]: needs javadoc

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java [hunk 5]
```diff
@@ -41,4 +41,11 @@ public ExhibitionAboutPage store(ExhibitionAboutPage exhibitionAboutPage) {
...
+        List<ExhibitionAboutPage> aboutPageList = findAll();
+        ExhibitionAboutPage exhibitionAboutPage = aboutPageList != null && !aboutPageList.isEmpty() ? aboutPageList.get(0):new ExhibitionAboutPage();
+        return exhibitionAboutPage;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:48]: merge with lline 47

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import edu.asu.diging.vspace.core.model.impl.ExhibitionAboutPage;
+import edu.asu.diging.vspace.core.services.IExhibitionAboutPageManager;
+
+@Controller
+public class ExhibitionPublicAboutPageController {
+    
+    Logger logger = LoggerFactory.getLogger(ExhibitionPublicAboutPageController.class);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java:16]: should be private

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    
+    @Autowired
+    private IExhibitionAboutPageManager aboutPageManager;
+    
+    @RequestMapping(value = "/exhibit/aboutPage")
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java:21]: should be `/exhibit/about`

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionModuleController.java [hunk 8]
```diff
@@ -29,6 +34,9 @@ public String module(@PathVariable("id") String id, @PathVariable("spaceId") Str
         if (space == null) {
             return "redirect:/exhibit/404";
         }
+        ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+        model.addAttribute("aboutPageTitle", exhibitionAboutPage.getTitle());
+        model.addAttribute("aboutPageText", exhibitionAboutPage.getAboutPageText());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionModuleController.java:39]: is that needed here? about page is its own page, isn't it?

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionSlideController.java [hunk 9]
```diff
@@ -69,6 +74,9 @@ public String slide(Model model, @PathVariable("slideId") String slideId, @PathV
             model.addAttribute("message", "Sorry, module has not been configured yet.");
             return "/exhibition/module";
         }
+        ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+        model.addAttribute("aboutPageTitle", exhibitionAboutPage.getTitle());
+        model.addAttribute("aboutPageText", exhibitionAboutPage.getAboutPageText());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionSlideController.java:79]: see above

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionSpaceController.java [hunk 10]
```diff
@@ -59,6 +64,9 @@ public String space(@PathVariable("id") String id, Model model) {
         if (space.getSpaceStatus() == null || space.getSpaceStatus().equals(SpaceStatus.PUBLISHED)
                 || authenticationFacade.getAuthenticatedUser() != null) {
             IExhibition exhibition = exhibitManager.getStartExhibition();
+            ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+            model.addAttribute("aboutPageTitle", exhibitionAboutPage.getTitle());
+            model.addAttribute("aboutPageText", exhibitionAboutPage.getAboutPageText());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionSpaceController.java:69]: see above

File: vspace/src/main/webapp/resources/extra/Home.css [hunk 11]
```diff
@@ -549,13 +564,31 @@
...
 }
 
+#aboutContent .closebtn {
+    position: absolute;
+    top: 8px;
+    right: 3px;
+    margin-left: 50px;
+}
+
+
```
Comment (jdamerow) [inline vspace/src/main/webapp/resources/extra/Home.css:591]: are these css changes still needed?

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html [hunk 12]
```diff
@@ -15,6 +15,17 @@
 					      	<div style="height: 5px; width: 5px; border-radius:50%; background-color: var(--success)"></div>
 					      	<em style="color: var(--dark-grey); font-size: 12px; margin-left: 10px">logged in as <span th:text="${#authentication.name}"></span></em>
 					    </div>
+		        	</li>
+		        	<a  href="/vspace/exhibit/aboutPage">
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:19]: this link won't work e.g. on staging since the app isn't called 'vspace' there. This needs to use thymeleaf's mechanism to create links.

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+package edu.asu.diging.vspace.web.exhibit.publicpages;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java:1]: I'm not a huge fan of the package names. I don't have a suggestion for better subpackage names than view and pages, but maybe at least drop the public part (since the `exhibit` part is intended to convey that)?
`edu.asu.diging.vspace.web.exhibit.pages`
`edu.asu.diging.vspace.web.exhibit.view`

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionModuleController.java (hunk 8 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionModuleController.java:39]: This is to allow the staff to edit the about page on the staff site, so i am taking the about page content and providing it here to allow them to edit. Is this okay?

Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicview/ExhibitionModuleController.java:39]: On the exhibition pages(module, space and slides) the navbar on top will only show the about page button if it is setup by the staff. Line 37-39 is there to check th:if if they exist in order to show the button. Is this okay? 
image

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java (hunk 13 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java:1]: Changed

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html (hunk 12 above)
Comment (glend17) [reply vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:19]: Changed

File: vspace/src/main/webapp/resources/extra/Home.css (hunk 11 above)
Comment (glend17) [reply vspace/src/main/webapp/resources/extra/Home.css:591]: No, removed

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java (hunk 7 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java:21]: Done

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java (hunk 6 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/publicpages/ExhibitionPublicAboutPageController.java:16]: Done

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java (hunk 5 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:48]: Done

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java (hunk 4 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:45]: Done

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/ExhibitionAboutPage.java [hunk 14]
```diff
@@ -52,5 +56,18 @@ public String getAboutPageText() {
...
     }
-
+    
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.diging.vspace.core.model.impl.ITextBlock#htmlRenderedText()
+     */
+    @Transient
```
[the remark this answers is not in the document]
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/ExhibitionAboutPage.java:66]: Done

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java [hunk 15]
```diff
@@ -41,4 +41,14 @@ public ExhibitionAboutPage store(ExhibitionAboutPage exhibitionAboutPage) {
         return repo.save(exhibitionAboutPage);
     }
     
+    /**
+     *  This method returns the Exhibition About Page set by a Staff, if empty
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:45]: This is actually not true. It returns the first page it finds.

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionModuleController.java [hunk 16]
```diff
@@ -29,6 +34,9 @@ public String module(@PathVariable("id") String id, @PathVariable("spaceId") Str
         if (space == null) {
             return "redirect:/exhibit/404";
         }
+        ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+        model.addAttribute("aboutPageTitle", exhibitionAboutPage.getTitle());
+        model.addAttribute("aboutPageText", exhibitionAboutPage.getAboutPageText());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionModuleController.java:39]: why is that here?

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSlideController.java [hunk 17]
```diff
@@ -69,6 +74,9 @@ public String slide(Model model, @PathVariable("slideId") String slideId, @PathV
             model.addAttribute("message", "Sorry, module has not been configured yet.");
             return "/exhibition/module";
         }
+        ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+        model.addAttribute("aboutPageTitle", exhibitionAboutPage.getTitle());
+        model.addAttribute("aboutPageText", exhibitionAboutPage.getAboutPageText());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSlideController.java:79]: same here, is that needed?

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSpaceController.java [hunk 18]
```diff
@@ -59,6 +64,9 @@ public String space(@PathVariable("id") String id, Model model) {
         if (space.getSpaceStatus() == null || space.getSpaceStatus().equals(SpaceStatus.PUBLISHED)
                 || authenticationFacade.getAuthenticatedUser() != null) {
             IExhibition exhibition = exhibitManager.getStartExhibition();
+            ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+            model.addAttribute("aboutPageTitle", exhibitionAboutPage.getTitle());
+            model.addAttribute("aboutPageText", exhibitionAboutPage.getAboutPageText());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSpaceController.java:69]: same as above

File: vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+	$("#mySidenav").css("width", "0");
+}
+
+
+
+ $(document).ready(function() {
+	 var divElement = document.getElementById("aboutPageDescription"); 
+	 divElement.innerHTML += aboutPageDescription;
+}); 
+</script>
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html:24]: why are you setting the content via javascript instead of directly in the html?

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html [hunk 20]
```diff
@@ -15,6 +15,17 @@
...
 					    </div>
+		        	</li>
+		        	<a  th:href="@{/exhibit/about}">
+		        	<span th:if="${aboutPageTitle!=null && !#strings.isEmpty(aboutPageTitle)}" class = "icon-info secondary" />     	
+	        		<span class="custom_Text"
+                                   th:if="${aboutPageTitle!=null && !#strings.isEmpty(aboutPageTitle)}"
+                                   th:text="About"
+                                   style="font-size: 18px; cursor: pointer; font-weight: bold;
+                                   margin-right:10px; margin-left:-10px; padding-right:10px; margin-top:10px"
+                                   >
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:26]: the about text should be on its own page

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html (hunk 19 above)
Comment (glend17) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html:24]: the aboutPageDescription will have the content like this, <p>hello<strong>How are you</strong>doing</p> as it is taken from the markdown editor. So i thought it is better to set it using javascript and append it to the inner html. Should I change this?

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html (hunk 20 above)
Comment (glend17) [reply vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:26]: This is the button to redirect to the About page which is its own page

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionModuleController.java (hunk 16 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionModuleController.java:39]: Removed

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSlideController.java (hunk 17 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSlideController.java:79]: removed

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSpaceController.java (hunk 18 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/view/ExhibitionSpaceController.java:69]: Removed

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java (hunk 15 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:45]: Changed

File: vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java [hunk 21]
```diff
@@ -87,6 +87,43 @@ public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
...
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        // If there is no exhibition, we go back to root url page.
+        if(exhibition==null) {
+            return "redirect:/";
+        }
+        //If no exhibition mode has been setup for existing exhibition, we skip modes and aspects.
+        if(exhibition.getMode() == null) {
+            return jp.proceed();
+        }
+        Map<IdPrefix, String> ids = getIds(args, signature);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java:105]: why is that here, is that needed?

File: vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java [hunk 22]
```diff
@@ -87,6 +87,43 @@ public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
...
+        // If exhibition is set to maintenance, set the default message.
+        if(exhibitionMode.equals(ExhibitionModes.MAINTENANCE)) {
+            ((Model) args[modelIndex]).addAttribute("modeValue", exhibitionMode.getValue());
+        }
+        // If user is not logged in and exhibition is not active, show maintenance page.
+        if(authFacade.getAuthenticatedUser()==null && !exhibitionMode.equals(ExhibitionModes.ACTIVE)) {
+            return "maintenance";
+        }
+        return jp.proceed();
+    }
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java:126]: this is pretty much a copy of the other method. should be more DRY.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java [hunk 23]
```diff
@@ -38,7 +46,20 @@
      */
     @Override
     public ExhibitionAboutPage store(ExhibitionAboutPage exhibitionAboutPage) {
+        IExhibition exhibition = exhibitionManager.getStartExhibition();
+        exhibition.setAboutPageConfigured(true);
+        exhibitionManager.storeExhibition((Exhibition)exhibition);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:51]: the above two lines only need to happen if `setAboutPageConfigured` returns false, right?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html (hunk 19 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html:52]: I don't really see a reason to use javascript here unless there is an issue with adding the html into the template html.

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html [hunk 24]
```diff
@@ -15,6 +15,17 @@
 					      	<div style="height: 5px; width: 5px; border-radius:50%; background-color: var(--success)"></div>
 					      	<em style="color: var(--dark-grey); font-size: 12px; margin-left: 10px">logged in as <span th:text="${#authentication.name}"></span></em>
 					    </div>
+		        	</li>
+		        	<a  th:href="@{/exhibit/about}">
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:19]: so this link will always be here even if there is no about page?

File: vspace/src/main/webapp/resources/extra/Home.css (hunk 11 above)
Comment (jdamerow) [reply vspace/src/main/webapp/resources/extra/Home.css:559]: pelase remove them completely so the file doesn't show up as changed anymore (see the checklist you checked of)

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java (hunk 22 above)
Comment (glend17) [reply vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java:95]: I have changed the method show exhibition, to show the methodName and added that as a parameter in redirect request, this is because when we hit about page on public site, there is no space or module that is set in the request parameters, so in line 131 it redirects to 404, this shouldnt be done if the requests targetmethod is to the `ExhibitionPublicAboutPageController` whose method is `showPublicAboutPage`

File: vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html (hunk 19 above)
Comment (glend17) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html:47]: Removed it

**Comment by glend17** [comment]
Closing the PR to fix unit test failure

**Comment by glend17** [comment]
Fixed unit test

File: vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java [hunk 25]
```diff
@@ -118,7 +128,7 @@ private Object redirectRequest(ProceedingJoinPoint jp, String spaceId, String mo
             return "maintenance";
         }
         // If the space and module Id is not found, show message on screen.
-        if(space==null && module==null) {
+        if(space==null && module==null && !methodName.equals(ABOUT_PAGE_PUBLIC_METHOD_NAME)) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java:131]: This will lead to bugs down the road. What if the controller method gets renamed, I guarantee this will be forgotten and it won't work anymore. Just don't put the controller for the about page in the exhibit package, so that the aspect never gets called. Or does it need anythings that's done in the aspect?

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java [hunk 26]
```diff
@@ -38,7 +43,22 @@
...
+        IExhibition exhibition = exhibitionManager.getStartExhibition();
+        if(!exhibition.isAboutPageConfigured()) {
+            exhibition.setAboutPageConfigured(true);
+            exhibitionManager.storeExhibition((Exhibition)exhibition);
+        }
         return repo.save(exhibitionAboutPage);
     }
     
+    /**
+     *  This method returns the first Exhibition About Page set by a Staff, if empty
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionAboutPageManager.java:55]: "if empty" doesn't mean much. what is empty? you mean if there is no exhibition object yet, don't you? Also, this needs to be proper javadoc (return value)

File: vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/pages/ExhibitionPublicAboutPageController.java [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+    
+    private Logger logger = LoggerFactory.getLogger(ExhibitionPublicAboutPageController.class);
+    
+    @Autowired
+    private IExhibitionAboutPageManager aboutPageManager;
+    
+    @RequestMapping(value = "/exhibit/about")
+    public String showPublicAboutPage(Model model) {
+        ExhibitionAboutPage exhibitionAboutPage = aboutPageManager.getExhibitionAboutPage();
+        model.addAttribute("exhibitionAboutPage", exhibitionAboutPage);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/exhibit/pages/ExhibitionPublicAboutPageController.java:24]: merge with line 23

File: vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html [hunk 28]
```diff
@@ -15,6 +15,17 @@
 					      	<div style="height: 5px; width: 5px; border-radius:50%; background-color: var(--success)"></div>
 					      	<em style="color: var(--dark-grey); font-size: 12px; margin-left: 10px">logged in as <span th:text="${#authentication.name}"></span></em>
 					    </div>
+		        	</li>
+		        	<a th:if="${aboutPageConfigured==true}"  th:href="@{/exhibit/about}">
+		        	<span th:if="${aboutPageConfigured==true}" class = "icon-info secondary" />     	
+	        		<span class="custom_Text"
+                                   th:if="${aboutPageConfigured==true}"
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/layouts/navbar.html:22]: Instead of have three if statements for each element I would simply surround all elements with a span and put the if in that span.

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java (hunk 25 above)
Comment (jdamerow) [reply vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java:131]: What about this comment?

**Comment by glend17** [comment]
Addressed review comments, raising PR

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- there should be just the 'i' icon (no text "About", that would be just when hovering)
- the text should be centered (why is there such a big margin from the left?)
image

**Comment by glend17** [comment]
Resolved PR comments, opening PR

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
The about page text is still not centered.

**Comment by glend17** [comment]
Centered the text, removed the menu button and added a back button

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- The back button is cut off.
- there should also be a back button on the top
- the back button should go back to the space the user was on when clicking on the about page icon.

**Comment by glend17** [comment]
addressed review comments

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+<body >
+
+<div layout:fragment="content"  >
+
+<div id="Module_1" class="main-contents" >
+     <div style = "float:left">
+     <a href="javascript:history.back()" class="btn primary-btn btn-large" style="float: left; ">Back</a>
+     </div>       
+		
+	<div style="justify-content:center; float:left">
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/aboutPagePublic.html:25]: why are the two divs float left? they are not meant to be next to each other. Also, give it a little bit more space between the heading and the button. This looks unnecessarily squished.

**Comment by glend17** [comment]
Re-opened pr with comments addressed
